sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/XmlMapperFactory.java

```diff
@@ -5,32 +5,88 @@
+    private final Object writeXmlDeclaration;
+    private final MethodHandle enableEmptyElementAsNull;
+    private final Object emptyElementAsNull;
+
     private static final String MUTABLE_COERCION_CONFIG = "com.fasterxml.jackson.databind.cfg.MutableCoercionConfig";
     private static final String COERCION_INPUT_SHAPE = "com.fasterxml.jackson.databind.cfg.CoercionInputShape";
     private static final String COERCION_ACTION = "com.fasterxml.jackson.databind.cfg.CoercionAction";
 
-    private MethodHandle coercionConfigDefaults;
-    private MethodHandle setCoercion;
-    private Object coercionInputShapeEmptyString;
-    private Object coercionActionAsNull;
-    private boolean useReflectionToSetCoercion;
+    private final MethodHandle coercionConfigDefaults;
+    private final MethodHandle setCoercion;
+    private final Object coercionInputShapeEmptyString;
+    private final Object coercionActionAsNull;
+    private final boolean useReflectionToSetCoercion;
 
     public static final XmlMapperFactory INSTANCE = new XmlMapperFactory();
 
     private XmlMapperFactory() {
         MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
 
+        MethodHandle createXmlMapperBuilder;
+        MethodHandle defaultUseWrapper;
+        MethodHandle enableWriteXmlDeclaration;
+        Object writeXmlDeclaration;
+        MethodHandle enableEmptyElementAsNull;
+        Object emptyElementAsNull;
+        try {
+            Class<?> xmlMapper = Class.forName(XML_MAPPER);
+            Class<?> xmlMapperBuilder = Class.forName(XML_MAPPER_BUILDER);
+            Class<?> fromXmlParser = Class.forName(FROM_XML_PARSER);
+            Class<?> toXmlGenerator = Class.forName(TO_XML_GENERATOR);
+
+            createXmlMapperBuilder = publicLookup.unreflect(xmlMapper.getDeclaredMethod("builder"));
+            defaultUseWrapper = publicLookup.unreflect(xmlMapperBuilder.getDeclaredMethod("defaultUseWrapper",
+                boolean.class));
+
+            enableWriteXmlDeclaration = publicLookup.unreflect(xmlMapperBuilder.getDeclaredMethod("enable",
+                Array.newInstance(toXmlGenerator, 0).getClass()));
+            writeXmlDeclaration = toXmlGenerator.getDeclaredField("WRITE_XML_DECLARATION").get(null);
+            enableEmptyElementAsNull = publicLookup.unreflect(xmlMapperBuilder.getDeclaredMethod("enable",
+                Array.newInstance(fromXmlParser, 0).getClass()));
+            emptyElementAsNull = fromXmlParser.getDeclaredField("EMPTY_ELEMENT_AS_NULL").get(null);
+        } catch (Throwable ex) {
+            // Throw the Error only if it isn't a LinkageError.
+            // This initialization is attempting to use classes that may not exist.
+            if (ex instanceof Error && !(ex instanceof LinkageError)) {
+                throw (Error) ex;
+            }
+
+            throw LOGGER.logExceptionAsError(new IllegalStateException("Failed to retrieve MethodHandles used to "
```

Comment:
wonder if we could add version helper string to linkage errors here (from `JacksonVersion`) - it was useful for environments like spark with limited access to logs